src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +18,42 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return phone;
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = cb.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            Predicate predicate = cb.and();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<Object> phonePredicate = cb.in(phoneRoot.get(entry.getKey()));
+                for (String parameter : entry.getValue()) {
+                    phonePredicate.value(parameter);
+                }
+                predicate = cb.and(predicate, phonePredicate);
+            }
+            query.where(predicate);
+            return session.createQuery(query).getResultList();
+        }
```

Comment:
It's worth creating a block catch. The message from Exception should be: 
`            throw new RuntimeException("Can't find all phones with those parameters: "
                    + stringParams + " Cause ", e);`

You can realize it within use the ArrayList (stringParams) and the method forEach(k, v) -> ....

k - key
v - value